### ISSUE TITLE
OIDC Callback HTTP Status Code 500

### ISSUE THREAD

**Issue by jameshearttech** [goharbor/harbor#18625]
If you are reporting a problem, please make sure the following information are provided:

**Expected behavior and actual behavior:**
I would expect that after the callback if the OIDC provider authorization succeeded I would be redirected to harbor with appropriate access.

**Steps to reproduce the problem:**

- Create a secret called config-overwrite-json with a key .data.config that contains json for CONFIG_OVERWRITE_JSON. Below is an example like my secret.

```json
{
  "auth_mode": "oidc_auth",
  "oidc_name": "Duo",
  "oidc_endpoint": "https://sso-abcdefgh.sso.duosecurity.com/oidc/5IHH6TAHFH28YCBSUEFW",
  "oidc_client_id": "5IHH6TAHFH28YCBSUEFW",
  "oidc_client_secret": "22Wd9HvzjLaOzQCvqfYan7cvmaZQ1k9OKKVL92wp",
  "oidc_scope": "profile,email",
  "oidc_verify_cert": "true",
  "oidc_auto_onboard": "true"
}
```

- Configure environment variable for harbor core by patching deployment with Kustomize. Below is example like my kustomization.yaml.

```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
patches:
  - patch: |-
      - op: add
        path: "/spec/template/spec/containers/0/env/-"
        value:
          name: CONFIG_OVERWRITE_JSON
          valueFrom:
            secretKeyRef:
              name: config-overwrite-json
              key: config
    target:
      group: apps
      version: v1
      kind: Deployment
      name: harbor-core
```

- Restart harbor core.
- Attempt login using OIDC.

I get through the SSO dialog in Duo. I see in the Duo logs my authorization was successful. After callback I get http status code 500 from harbor.

![Dev Tools Screenshot](https://user-images.githubusercontent.com/67252279/235488615-918e306e-7094-482a-afea-bb2c38d2f79a.png)

I see in the harbor core logs `Handler crashed with error interface conversion: interface {} is nil, not string`. That's the only thing that stands out to me.

```
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /usr/local/go/src/runtime/asm_amd64.s:1598
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /usr/local/go/src/net/http/server.go:1995
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /usr/local/go/src/net/http/server.go:2936
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /usr/local/go/src/net/http/server.go:2122
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /harbor/src/server/middleware/middleware.go:57
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /harbor/src/server/middleware/url/parse.go:22
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /usr/local/go/src/net/http/server.go:2122
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /harbor/src/server/middleware/middleware.go:57
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /harbor/src/server/middleware/mergeslash/mergeslash.go:17
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /usr/local/go/src/net/http/server.go:2122
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /harbor/src/server/middleware/trace/trace.go:28
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /usr/local/go/src/net/http/server.go:2122
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /harbor/src/server/middleware/metric/metric.go:60
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /usr/local/go/src/net/http/server.go:2122
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /harbor/src/server/middleware/middleware.go:57
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /harbor/src/server/middleware/requestid/requestid.go:46
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /usr/local/go/src/net/http/server.go:2122
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /harbor/src/server/middleware/middleware.go:57
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /harbor/src/server/middleware/log/log.go:33
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /usr/local/go/src/net/http/server.go:2122
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /harbor/src/server/middleware/session/session.go:35
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /usr/local/go/src/net/http/server.go:2122
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /harbor/src/server/middleware/middleware.go:57
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /harbor/src/server/middleware/csrf/csrf.go:68
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /harbor/src/vendor/github.com/gorilla/csrf/csrf.go:297
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /usr/local/go/src/net/http/server.go:2122
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.296 [C] [config.go:507]  /harbor/src/server/middleware/csrf/csrf.go:48
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /usr/local/go/src/net/http/server.go:2122
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /harbor/src/server/middleware/middleware.go:57
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /harbor/src/server/middleware/orm/orm.go:54
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /usr/local/go/src/net/http/server.go:2122
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /harbor/src/server/middleware/middleware.go:57
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /harbor/src/server/middleware/notification/notification.go:31
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /usr/local/go/src/net/http/server.go:2122
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /harbor/src/server/middleware/middleware.go:52
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /usr/local/go/src/net/http/server.go:2122
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /harbor/src/server/middleware/artifactinfo/artifact_info.go:62
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /usr/local/go/src/net/http/server.go:2122
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /harbor/src/server/middleware/middleware.go:57
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /harbor/src/server/middleware/security/security.go:62
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /usr/local/go/src/net/http/server.go:2122
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /harbor/src/server/middleware/middleware.go:57
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /harbor/src/server/middleware/security/security.go:75
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /usr/local/go/src/net/http/server.go:2122
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /harbor/src/server/middleware/middleware.go:52
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /harbor/src/vendor/github.com/beego/beego/v2/server/web/router.go:1002
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /harbor/src/vendor/github.com/beego/beego/v2/server/web/filter.go:83
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /harbor/src/vendor/github.com/beego/beego/v2/server/web/router.go:1233
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /usr/local/go/src/reflect/value.go:370
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /usr/local/go/src/reflect/value.go:586
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /harbor/src/core/controllers/oidc.go:110
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /harbor/src/pkg/oidc/helper.go:187
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /usr/local/go/src/runtime/iface.go:262
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:507]  /usr/local/go/src/runtime/panic.go:884
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:501]  Handler crashed with error interface conversion: interface {} is nil, not string
2023-05-01T05:10:00-10:00	2023/05/01 15:10:00.295 [C] [config.go:500]  the request url is  /c/oidc/callback
```

**Versions:**
- harbor: 2.8.0
- kubernetes: v1.26.4
- os: ubuntu 22.04.2 LTS
- kernel: 5.15.0-70-generic
- container-runtime: containerd://1.6.20

**Additional context:**

**Comment by stonezdj** [comment]
From the error message, it could be the oidc token doesn't contain the attribute `id_token`
Can you please try to configure the Harbor instance with this configuration directly?(without CONFIG_OVERWRITE_JSON) If it works, then we could screen out the invalid OIDC configuration.

**Comment by jameshearttech** [comment]
@stonezdj I took your advice and configured OIDC directly (i.e., in the UI). I was able to resolve the problem, which was a misconfiguration on my end. I had `"oidc_scope": "profile,email"`, which did not work because `oidc_scope` must include `openid`, which makes sense. I guess I assumed it was included by default. The docs do not mention it must be included, but perhaps it's understood to be common sense because it's required. Anyway, my issue is resolved so I'm closing this issue. Thanks!
